### PR TITLE
Remove all warning messages from `tests/integration_tests/test_sklearn.py`

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

A part of https://github.com/optuna/optuna/issues/3815.

## Description of the changes

As the title.

### CODE REVIEW

File: tests/integration_tests/test_sklearn.py [hunk 1]
```diff
@@ -367,7 +370,9 @@ def test_callbacks() -> None:
         callbacks=callbacks,  # type: ignore
     )
 
-    optuna_search.fit(X, y)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=ConvergenceWarning)
+        optuna_search.fit(X, y)
```
Comment (toshihikoyanase) [inline tests/integration_tests/test_sklearn.py:375]: [Note] The optimization result of `SGDClassifier` is out-of-scope of this unit test, and the change sounds reasonable.

```console
python3.8/site-packages/sklearn/linear_model/_stochastic_gradient.py:705: ConvergenceWarning: Maximum number of iteration reached before convergence. Consider increasing max_iter to improve the fit.
    warnings.warn(
```

File: tests/integration_tests/test_sklearn.py [hunk 2]
```diff
@@ -308,6 +310,7 @@ def test_objective_error_score_invalid() -> None:
 
 # TODO(himkt): Remove this method with the deletion of deprecated distributions.
 # https://github.com/optuna/optuna/issues/2941
+@pytest.mark.filterwarnings("ignore::FutureWarning")
```
Comment (toshihikoyanase) [inline tests/integration_tests/test_sklearn.py:313]: In this test function, old distributions such as `UniformDistribution` and `LogUniformDistribution`, and supressing `FututreWarning` sounds reasonable.

**Review by toshihikoyanase — APPROVED** [review]
LGTM. Thank you!

**Comment by toshihikoyanase** [comment]
@knshnb Could you review this PR when you have time?

**Review by knshnb — APPROVED** [review]
LGTM. Thanks for the PR!

**Pull request merged**